sopel/modules/currency.py

```diff
@@ -2,109 +2,196 @@
+
+    if of == 'BTC' or to == 'BTC':
+        if of == 'BTC':
+            code = to
+            reverse = False
+        else:
+            code = of
+            reverse = True
+
+        return btc_rate(code, reverse)
+
+    if of not in rates_fiat_json['rates']:
+        return "Sorry {} isn't currently supported".format(of)
+
+    if to not in rates_fiat_json['rates']:
+        return "Sorry {} isn't currently supported".format(to)
+
+    return (1 / rates_fiat_json['rates'][of]) * rates_fiat_json['rates'][to]
+
+
+def exchange(bot, match):
     """Show the exchange rate between two currencies"""
-    if not trigger.group(2):
-        return bot.reply("No search term. An example: .cur 20 EUR in USD")
-    match = regex.match(trigger.group(2))
     if not match:
-        # It's apologetic, because it's using Canadian data.
         bot.reply("Sorry, I didn't understand the input.")
         return NOLIMIT
 
-    amount, of, to = match.groups()
+    try:
+        update_rates(bot)  # Try and update rates. Rate-limiting is done in update_rates()
+    except requests.exceptions.RequestException as e:
+        bot.reply("Something went wrong while I was getting the exchange rate.")
+        LOGGER.error("Error in GET request: {}".format(e))
+        return NOLIMIT
+    except ValueError:
+        bot.reply("Error: Got malformed data.")
+        LOGGER.error("Invalid json on update_rates")
+        return NOLIMIT
+
+    query = match.string
+
+    others = query.split()
+    amount = others.pop(0)
+    of = others.pop(0)
+    others.pop(0)
+
+#    amount, of, _, *others = query.split() # I'd much rather use this, but it's not python 2.7 compatible
+
     try:
         amount = float(amount)
     except ValueError:
         bot.reply("Sorry, I didn't understand the input.")
     except OverflowError:
         bot.reply("Sorry, input amount was out of range.")
-    display(bot, amount, of, to)
 
+    out_string = '{} {} is'.format(amount, of.upper())
+    for to in others:
+        try:
+            out_string = build_reply(bot, amount, of.upper(), to.upper(), out_string)
+        except ValueError:
+            return NOLIMIT
 
-def display(bot, amount, of, to):
-    if not amount:
-        bot.reply("Zero is zero, no matter what country you're in.")
-    try:
-        of_rate, of_name = get_rate(of)
-        if not of_name:
-            bot.reply("Unknown currency: %s" % of)
-            return
-        to_rate, to_name = get_rate(to)
-        if not to_name:
-            bot.reply("Unknown currency: %s" % to)
-            return
-    except Exception:  # TODO: Be specific
-        bot.reply("Something went wrong while I was getting the exchange rate.")
-        return NOLIMIT
+    bot.reply(out_string[0:-1])
+
+
+@commands('cur', 'currency', 'exchange')
+@example('.cur 100 usd in btc cad eur')
+def exchange_cmd(bot, trigger):
+    if not trigger.group(2):
+        return bot.reply("No search term. An example: .cur 100 usd in btc cad eur")
```

Comment:
We're trying _really_ hard to fix the hard-coded `.`s in usage messages all through the core modules!

```suggestion
        return bot.reply("No search term. An example: {}cur 100 usd in btc cad eur"
                         .format(bot.config.core.help_prefix))
```

Optionally, also replace "An example" with "Usage" to align with most of the other core command set.